src/meta-srv/src/selector/weighted_choose.rs

```diff
@@ -92,6 +92,10 @@ where
     }
 
     fn choose_multiple(&mut self, amount: usize) -> Result<Vec<Item>> {
+        if amount >= self.items.len() {
+            return Ok(self.items.iter().map(|item| item.item.clone()).collect());
+        }
+
```

Comment:
Respect the interface comment: https://github.com/GreptimeTeam/greptimedb/blob/f3a1e699ec358c980cdac2904f6fa99991eae7c2/src/meta-srv/src/selector/weighted_choose.rs#L29